assets/chef-recipes/cookbooks/suse-connect/libraries/registration.rb

```diff
@@ -24,16 +24,32 @@ def self.load_credentials
       # Generate registration command for the machine
       # @param suse_connect_parameters [Hash] subscription parameters in format {'email' : String, 'key' : String, 'registration_proxy_url' : String }
       def self.register_node_command(suse_connect_parameters)
-        # Copying configuration script from the registration proxy requires HTTP protocol
-        http_proxy_url = suse_connect_parameters['registration_proxy_url'].sub('https://', 'http://')
-        "curl #{URI.join(http_proxy_url, '/tools/rmt-client-setup')} --output rmt-client-setup; yes | sh rmt-client-setup #{suse_connect_parameters['registration_proxy_url']}; SUSEConnect -r #{suse_connect_parameters['key']} -e #{suse_connect_parameters['email']} --url #{suse_connect_parameters['registration_proxy_url']}"
-      end
+        if use_proxy?(suse_connect_parameters)
+          # Copying configuration script from the registration proxy requires HTTP protocol
+          http_proxy_url = suse_connect_parameters['registration_proxy_url'].sub('https://', 'http://')
+          return "curl #{URI.join(http_proxy_url, '/tools/rmt-client-setup')} --output rmt-client-setup; yes | sh rmt-client-setup #{suse_connect_parameters['registration_proxy_url']}; SUSEConnect -r #{suse_connect_parameters['key']} -e #{suse_connect_parameters['email']} --url #{suse_connect_parameters['registration_proxy_url']}"
```

Comment:
Format this string as multiline, so each separate command will be on a separate line.